src/components/Steps/Step/Step.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import cx from 'classnames'
+import Icon from '@santiment-network/ui/Icon'
+import styles from './Step.module.scss'
+
+const Step = ({
+  className,
+  active,
+  status = 'wait',
+  stepNumber,
+  disabled,
+  description,
+  title,
+  icons,
+  stepIndex,
+  onStepClick,
+  onClick,
+  size,
+  ...restProps
+}) => {
+  const handleClick = (...args) => {
+    if (onClick) {
+      onClick(...args)
+    }
+
+    onStepClick(stepIndex)
+  }
+
+  const renderIconNode = () => {
+    if (icons && icons.process && status === 'process') {
+      return icons.process
+    }
+    if (status === 'finish' && !disabled) {
+      return <Icon type='checkmark' />
+    }
+    return <span className={styles.stepsIcon}>{stepNumber}</span>
+  }
+
+  const getClassString = () => {
+    if (disabled) {
+      return styles.stepsItemDisabled
+    }
+    if (status === 'wait' && !disabled) {
+      return styles.stepsItemWait
+    }
+    if (status === 'process') {
+      return styles.stepsItemProcess
+    }
+    if (status === 'finish' && !disabled) {
+      return styles.stepsItemFinish
+    }
+  }
```

Comment:
1. Same as the `[D-0-1]`;
2. `status` and `disabled` can be passed as arguments so the function can be moved outside the component render;
3. `&& !disabled` is redundant since the first `if (disabled)` contains return;
4. Rename classes so the `styles` object can act as a hash table: `styles[status]`.

The result: `disabled ? styles.disabled : styles[status]`